### DISCUSSION TITLE
Unable to alter margins in base R boxplot

### DISCUSSION THREAD

**Discussion by kubu4** [RobertsLab/resources#1344] [Q&A] [answered]
Firstly, here's the R Project I'm working with:

https://github.com/epigeneticstoocean/2018_L18-adult-methylation/tree/main/code/ballgown_diff-isoform-expression

If you download that entire directory, you'll be able to reproduce everything. NOTE: You'll need to install `ballgown` from BioConductor (good luck!)...

So, if I generate a box plot using the following code, here's what it looks like:

```r
# Pull all transcript expression values stored in FPKM measurement from ballgown object
fpkm <- texpr(bg, meas = "FPKM")

# Log transform data and add 1 to all values to artificually prevent log2(0).
fpkm <-log2(fpkm+1)

# Set plot colors for F/M
myColors <- ifelse(phenotype_table$Sex=="F", rgb(0.8,0.1,0.3,0.6),
                   ifelse(phenotype_table$Sex=="M", rgb(0.1,0.1,0.7,0.5),
                          "grey90")
)

# Generate box plot.
boxplot(fpkm, col = myColors, las=2, ylab='log2(FPKM+1)')
```

![initial R boxplot](https://user-images.githubusercontent.com/4514104/142672964-ee3d188c-d05f-4a3d-b9ff-0e6201013fa5.png)

---

Now, if I want to edit the margins so that I can see the full text on the x-axis, I add this line and here's how the plot looks now:

```r
# Set box plot margins (extreme values to show no effect on boxplot)
par(mar=c(100,100,100,100))
# Generate box plot.
boxplot(fpkm, col = myColors, las=2, ylab='log2(FPKM+1)')
```

![R boxplot with modified margin values is no different than default](https://user-images.githubusercontent.com/4514104/142673309-89b0b18b-2d25-43d8-b794-3ac7ae72d174.png)

---

So, anyone have any ideas on how to resolve this?

The alternative (and, arguably better approach), is to use `ggplot2`, but I'm not sure where to even start. Some of my feeble attempts have led to nothing but errors (`Error: `data` must be a data frame, or other object coercible by `fortify()`, not a numeric vector.`) or (if I convert the `fpkm` object to a data frame) lack of understanding on how to use the `aes()` aspect of `ggplot()`.

Would appreciate any/all assistance!

**Comment by kubu4** [comment]
And, here's a link to a tab-delimited version of the `ballgown` object I am working with:

https://github.com/epigeneticstoocean/2018_L18-adult-methylation/blob/main/code/ballgown_diff-isoform-expression/analyses/fpkm_all_samples.tab

**Comment by sr320** [comment, marked as answer]
Done in tidyverse

https://github.com/epigeneticstoocean/2018_L18-adult-methylation/blob/main/code/05-ballgown-boxplots.md

**Reply by kubu4** [reply]
Just adding written description for posterity.

Didn't resolve the actual base R boxplot margin issue.

Instead, went with `ggplot()`. This also required using `pivot_longer()` to rotate the data frame for R to be happy with it.
